pallets/inflation-manager/src/lib.rs

```diff
@@ -180,28 +171,34 @@ pub mod pallet {
 			let inflation_config = InflationConfiguration::<T>::get();
 			let mut inflation_parameters = InflationParameters::<T>::get();
 
-			// if we're at the end of a year
+			// if we're at the end of a year or initializing inflation
 			if now >= target_block {
 				// update current year
 				CurrentYear::<T>::put(new_year);
 
-				// check if we need to recalculate inflation parameters for a new year
-				// update inflation parameters if we havent reached the stagnation year
+				// if we're at DoInitializeAt, then we need to adjust total issuance for delayed TGE
+				if now == T::DoInitializeAt::get() {
```

Comment:
Do I understand right only when the T::DoRecalculationAt == T::DoInitializeAt, we'll do the fund difference?

Reply:
Yup, the on_finalize_hook works as usual, with the only change being that if the current block is where delayed TGE takes affect, it will fund the pot.

Reply:
Ahhh... Got it. If the DoInitializeAt is 0, we shouldn't do the fund diff. If it's set up as the number, the DoRecalculationAt will set up the same value because of the initialized delayed inflation. 

I have to know the relationship between the DoInitializeAt and DoRecalculationAt at the initialized delay inflation to understand the behavior. Therefore, could you add one comment to make the relation explicit?

Reply:
initialize_delay_inflation function will take DoInitializeAt as argument, then set DoRecalculationAt, the storage value, as this argument, so when on_finalize hook runs as usual, it will see that if DoRecalculationAt has been reached, it will calculate inflation parameters and block rewards as usual, and in this hook, if DoRecalculationAt==DoInitializeAt, it will also fund the diff. In delayed TGE case, DoRecalculationAt will be the block where inflation is initialized.
This way the hook remains the same with only very minimal changes.